File: semester_2/Other/reverse.c
#include <stdio.h>

int reverse (int n);


int main(){
    int num;

    printf("Enter an integer: ");
    if (scanf("%d",&num) != 1){
        printf("Error: Enter a valid integer");
        return 1;
    }

    int result = reverse(num);
    printf("The reversed number is: %d", result);
    return 0;
}

int reverse (int n){
    int r,sum=0;
    while(n != 0){
    r = n %10;
    sum = sum*10 + r;
    n = n/10;}

    return sum;
}

//add a function for negative inputs ? + also do error handling
//r = -8 % 10 = -8 so my code already works because it is in c for -ve numbers BUT it wont for other languages
//negative numbers should be explicitly handled eg: python
// % only works for integers sooo floating point numbers will not be reversed

File: semester_2/Other/reverse.py
import math

def reverse(n):
    sum = 0
    negative = False  # define this before otherwise when n is +ve if block generates an NameError
    if (n < 0):
        negative = True
        n = -n
    while(n != 0):
        r = n % 10
        sum = sum*10 + r
        n = n // 10   # integer division
    if(negative):
        sum = -sum
    return sum

user_input = input("Enter a number: ")

try:
    num = int(user_input)  # convert input to integer (works with negatives too)
    print("Valid integer!")
    
    result = reverse(num)

    print(f"The reversed number is: {result}") #f inside is necessary

except ValueError:
    print("Invalid input. Please enter an integer.")





